astropy/coordinates/frame_attributes.py

```diff
@@ -420,6 +421,69 @@ def convert_input(self, value):
                 transformedobj = transformedobj.frame
             return transformedobj, True
 
+class DifferentialFrameAttribute(FrameAttribute):
+    """A frame attribute which is a differential instance.
+
+    The optional ``allowed_classes`` argument allows specifying a restricted
+    set of valid differential classes to check the input against. Otherwise,
+    any `~astropy.coordinates.BaseDifferential` subclass instance is valid.
+
+    Parameters
+    ----------
+    default : object
+        Default value for the attribute if not provided
+    allowed_classes : tuple, optional
+        A list of allowed differential classes for this attribute to have.
+    secondary_attribute : str
+        Name of a secondary instance attribute which supplies the value if
+        ``default is None`` and no value was supplied during initialization.
+    """
+    def __init__(self, default=None, allowed_classes=None,
+                 secondary_attribute=''):
+
+        if allowed_classes is not None:
+            self.allowed_classes = tuple(allowed_classes)
+        else:
+            self.allowed_classes = None
```

Comment:
Maybe just set to `BaseDifferential`? That makes the check below more straightforward.